src/System.Net.Security/src/System/Net/Unix/CertModule.cs

```diff
@@ -132,10 +132,33 @@ internal override X509Certificate2 GetRemoteCertificate(SafeDeleteContext securi
         //
         internal override string[] GetRequestCertificateAuthorities(SafeDeleteContext securityContext)
         {
-            // TODO (Issue #3362) populate issuers
-            string[] issuers = Array.Empty<string>();          
+            using (SafeSharedX509NameStackHandle names = Interop.libssl.SSL_get_client_CA_list(securityContext.SslContext))
+            {
+                if (names.IsInvalid)
+                {
+                    return Array.Empty<string>();
+                }
+
+                int nameCount = Interop.Crypto.GetX509NameStackFieldCount(names);
+
+                if (nameCount == 0)
+                {
+                    return Array.Empty<string>();
+                }
 
-            return issuers;
+                string[] clientAuthorityNames = new string[nameCount];
+
+                for (int i = 0; i < nameCount; i++)
+                {
+                    using (SafeSharedX509NameHandle nameHandle = Interop.Crypto.GetX509NameStackField(names, i))
+                    {
+                        X500DistinguishedName dn = Interop.Crypto.LoadX500Name(nameHandle);
```

Comment:
One doubt about this name. Will it match the chain.ChainElements[].Certificate.Issuer field? I am trying to confirm that the name returned here would pass the checks at around line 515 in _SecureChannel.cs